bot/cogs/doc.py

```diff
@@ -0,0 +1,546 @@
+import asyncio
+import functools
+import logging
+import random
+import re
+from collections import OrderedDict
+from ssl import CertificateError
+from typing import Dict, List, Optional, Tuple
+
+import discord
+from aiohttp import ClientConnectorError
+from bs4 import BeautifulSoup
+from discord.ext import commands
+from markdownify import MarkdownConverter
+from requests import ConnectionError
+from sphinx.ext import intersphinx
+
+from bot.constants import (
+    ADMIN_ROLE, ERROR_REPLIES, MODERATOR_ROLE,
+    OWNER_ROLE, SITE_API_KEY, SITE_API_URL
+)
+from bot.decorators import with_role
+from bot.exceptions import CogBadArgument
+
+
+log = logging.getLogger(__name__)
+logging.getLogger('urllib3').setLevel(logging.WARNING)
+
+
+UNWANTED_SIGNATURE_SYMBOLS = ('[source]', '¶')
+WHITESPACE_AFTER_NEWLINES_RE = re.compile(r"(?<=\n\n)( +)")
```

Comment:
Use `\s` rather than ` `

Reply:
rather than...?

Reply:
whoops, looks like using \` on a space doesn't work. Whitespace is defined as \s, and documentation may contain tabs or zero-width spaces or similar (happens on a couple ones, actually), so instead of `( +)` you should use `(\s+)`